modules/tspec/internal/formatters/fmt.go

```diff
@@ -0,0 +1,104 @@
+package formatters
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/cucumber/messages-go/v10"
+	"github.com/gruntwork-io/terratest/modules/tspec/colors"
+	"github.com/gruntwork-io/terratest/modules/tspec/internal/models"
+	"github.com/gruntwork-io/terratest/modules/tspec/internal/utils"
+)
+
+var (
+	red    = colors.Red
+	redb   = colors.Bold(colors.Red)
+	green  = colors.Green
+	blackb = colors.Bold(colors.Black)
+	yellow = colors.Yellow
+	cyan   = colors.Cyan
+	cyanb  = colors.Bold(colors.Cyan)
+	whiteb = colors.Bold(colors.White)
+)
+
+// repeats a space n times
+var s = utils.S
+
+var (
+	passed    = models.Passed
+	failed    = models.Failed
+	skipped   = models.Skipped
+	undefined = models.Undefined
+	pending   = models.Pending
+)
+
+type sortFeaturesByName []*models.Feature
+
+func (s sortFeaturesByName) Len() int           { return len(s) }
+func (s sortFeaturesByName) Less(i, j int) bool { return s[i].Feature.Name < s[j].Feature.Name }
+func (s sortFeaturesByName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+type sortPicklesByID []*messages.Pickle
+
+func (s sortPicklesByID) Len() int { return len(s) }
+func (s sortPicklesByID) Less(i, j int) bool {
+	iID := mustConvertStringToInt(s[i].Id)
+	jID := mustConvertStringToInt(s[j].Id)
+	return iID < jID
+}
+func (s sortPicklesByID) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+type sortPickleStepResultsByPickleStepID []models.PickleStepResult
+
+func (s sortPickleStepResultsByPickleStepID) Len() int { return len(s) }
+func (s sortPickleStepResultsByPickleStepID) Less(i, j int) bool {
+	iID := mustConvertStringToInt(s[i].PickleStepID)
+	jID := mustConvertStringToInt(s[j].PickleStepID)
+	return iID < jID
+}
+func (s sortPickleStepResultsByPickleStepID) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+func mustConvertStringToInt(s string) int {
+	i, err := strconv.Atoi(s)
+	if err != nil {
+		panic(err)
+	}
+
+	return i
+}
+
+// DefinitionID ...
+func DefinitionID(sd *models.StepDefinition) string {
+	ptr := sd.HandlerValue.Pointer()
+	f := runtime.FuncForPC(ptr)
+	file, line := f.FileLine(ptr)
+	dir := filepath.Dir(file)
+
+	fn := strings.Replace(f.Name(), dir, "", -1)
```

Comment:
Struggling to understand this code too. In part, due to lots of one and two letter variable names, and in part, as I'm missing some context of how this BDD API is defined...